dla_cnn/desi/model_v2.1.py

```diff
@@ -0,0 +1,243 @@
+        return tf.nn.avg_pool2d(input=h_conv, ksize=[1, pool_kernel, 1, 1], strides=[1, pool_stride, 1, 1], padding='SAME')
+
+def variable_summaries(var, name, collection):
+    """Attach a lot of summaries to a Tensor."""
+    writer = tf.summary.create_file_writer("summary_file")
+    #tf.summary.experimental.set_step()
+    with tf.name_scope('summaries') as r:
+        mean = tf.reduce_mean(input_tensor=var)
+        with writer.as_default():
+            tf.summary.scalar('mean/' + name,mean,step=step)
+        with tf.name_scope('stddev'):
+            stddev = tf.sqrt(tf.reduce_mean(input_tensor=tf.square(var - mean)))
+        with writer.as_default():
+            tf.summary.scalar('stddev/' + name, stddev,step=step)
+            tf.summary.scalar('max/' + name, tf.reduce_max(input_tensor=var),step=step)
+            tf.summary.scalar('min/' + name, tf.reduce_min(input_tensor=var),step=step)
+            tf.summary.histogram(name, var)
+       
+
+     """
+    every tf.summary.scalar command needs a new parameter "step" but I have no idea how to pass value to this parameter,
+    maybe use tf.summary.experimental.set_step()? 
+    the parameter "collection" is not used, I changed this part using writer.as_default(), but I have no idea whether I keep the logic correct
+     """
+
+        
+
+def build_model(hyperparameters):
+    """
+    This is Model_v5 from SDSS
+
+    Parameters
+    ----------
+    hyperparameters
+
+    Returns
+    -------
+
+●   should use the hyperparameters in dla_cnn/models/model_gensample_v7.1_hyperparams.json
+
+    """
+    learning_rate = hyperparameters['learning_rate']
+    batch_size = hyperparameters['batch_size']
+    l2_regularization_penalty = hyperparameters['l2_regularization_penalty']
+    fc1_n_neurons = hyperparameters['fc1_n_neurons']
+    fc2_1_n_neurons = hyperparameters['fc2_1_n_neurons']
+    fc2_2_n_neurons = hyperparameters['fc2_2_n_neurons']
+    fc2_3_n_neurons = hyperparameters['fc2_3_n_neurons']
+    conv1_kernel = hyperparameters['conv1_kernel']
+    conv2_kernel = hyperparameters['conv2_kernel']
+    conv3_kernel = hyperparameters['conv3_kernel']
+    conv1_filters = hyperparameters['conv1_filters']
+    conv2_filters = hyperparameters['conv2_filters']
+    conv3_filters = hyperparameters['conv3_filters']
+    conv1_stride = hyperparameters['conv1_stride']
+    conv2_stride = hyperparameters['conv2_stride']
+    conv3_stride = hyperparameters['conv3_stride']
+    pool1_kernel = hyperparameters['pool1_kernel']
+    pool2_kernel = hyperparameters['pool2_kernel']
+    pool3_kernel = hyperparameters['pool3_kernel']
+    pool1_stride = hyperparameters['pool1_stride']
+    pool2_stride = hyperparameters['pool2_stride']
+    pool3_stride = hyperparameters['pool3_stride']
+    pool1_method = 1        # Removed from hyperparameter search because it never chose anything but this method
+    pool2_method = 1
+    pool3_method = 1
+
+    INPUT_SIZE = 400 # the pixel of input image
```

Comment:
Let's also make this a `hyperparameter` or an input variable.

Reply:
ok,I have made it  a hyperparameter, but this need to add a new hyperparameter "INPUT_SIZE"in model_gensample_v7.1_hyperparams.json